keps/sig-scalability/0001-enhance-network-performance-benchmark-testcases.md

```diff
@@ -0,0 +1,188 @@
+
+## Summary
+
+We are proposing to add more testcases for network performance benchmarking.
+
+## Motivation
+
+As Kubernetes is increasingly being adopted by various telco cloud providers for deployments, 
+it brings additional networking requirements/scenarios to be supported.  
+With wide array of network functions being containerized, the need to analyse and 
+monitor the networking performance metrics such as latency, throughput etc has arisen.
+
+We have come across 3 test frameworks which have limited testcases for network performance benchmarking.
+
+a) E2E (test\e2e\network\networking_perf.go): This framework has only 2 testcases for measuring throughput.
+
+b) netperf (https://github.com/kubernetes/perf-tests/tree/master/network/benchmarks/netperf) : This framework 
+   measures only the throughput between client-server pods for various MSS(Maximum Segment size). 
+   Other performance metrics such as latency , loss-rate , response time are not measured.
+   
+c) clusterloader2 (https://github.com/kubernetes/perf-tests/tree/master/clusterloader2) : This framework 
+   measures only network latency. Other performance metrics are not measured.
+
+### Goals
+
+The primary goal of this proposal is to enhance the testcases to cover various 
+networking scenarios :
+
+a) IPv4 , IPv6 , Dual-Stack.
+
+b) Protocols: TCP, UDP, HTTP/HTTPS.
+
+c) Pod-to-Pod , with services ClusterIP , NodePort etc.
+
+We intend to submit tests to measure the following network performance metrics for various 
+scenarios such as different services (ClusterIP/NodePort), with IPv4/IPv6 configured etc.
+
+a)  Throughput (For TCP traffic)
+
+b)  Packets per second (For UDP traffic)
+
+c)  Loss rate (For UDP traffic)
+
+d)  Latency/Jitter (For UDP/HTTP/HTTPS)
+    (Note: Latency being measured is a one-way latency)
+
+e)  Response time (For HTTP/HTTPs)
+    (Note: Response time is a two-way latency plus processing time)
+
+Our important goal is to make sure these benchmarking testcases are run periodically,
+so that any degradation in performance is detected in early phase of feature delivery.
+
+### Non-goals
+
+Benchmarking and measurements with intermediate services like ingress controllers, load-balancers 
+are out of scope of this proposal.
+
+### Challenges and Open Questions [optional]
+
+1) Which is the suitable hosting framework for proposed tests ? 
+2) Periodic execution of these proposed tests ?
+
+
+## Proposal
+
+### Test Scenarios
+
+Detailed test scenarios are listed in below link
+https://docs.google.com/document/d/1_760ptAHBY_c56slWhjl6PJQOrf__vQFJXkL58ewg1M/edit?usp=sharing
+
+Test Scenarios capture following aspects :
+
+a) Point-to-Point(P2P), Services.
+
+b) Client-Server POD location on worker node (co-located , distributed).
+
+c) Client-Server POD ratio (1:1 , N:1 , N:M)
+
+### Tools
+
+We intend to use the following tools for testing :
+
+a) iperf3: For TCP throughput measurement.
+
+b) iperf2: For UDP latency, loss rate , pps measurement.
```

Comment:
is it possible to just use only one to avoid confusions?

Reply:
iperf3 is a bandwidth tester tool (https://github.com/esnet/iperf/issues/547).
where as , iperf2 provides in enhanced option to measure latency (https://www.reddit.com/r/networking/comments/8j3rky/iperf_enhanced_reports_for_udp_latency/). 

Hence we can't use the same tool for both.

Reply:
I wanted to mean that iperf2 measured bandwidth too, and maybe it would be simpler using one tool only for all tests.
Anyway, is up to you, was mostly curiosity